llvm/test/Transforms/CheriBoundAllocas/broken-phi.ll

```diff
@@ -0,0 +1,59 @@
+; in the CheriBoundAllocas pass.
+; PHI node has multiple entries for the same basic block with different incoming values!
+; FIXME: remove -disable-verify+not --crash once fixed
+; REQUIRES: asserts
+; RUN: opt -cheri-bound-allocas -S -disable-verify < %s | FileCheck %s
+; RUN: not --crash opt -cheri-bound-allocas -S < %s
+target datalayout = "e-m:e-pf200:128:128:128:64-p:64:64-i64:64-i128:128-n64-S128-A200-P200-G200"
+target triple = "riscv64-unknown-freebsd13"
+
+%class.QString = type { %struct.QTypedArrayData addrspace(200)* }
+%struct.QTypedArrayData = type { %struct.QArrayData }
+%struct.QArrayData = type { i32, i32, i32, i8 addrspace(200)* }
+
+declare %class.QString addrspace(200)* @use(%class.QString addrspace(200)* %arg)
+
+define hidden void @_ZNK7KScreen6Doctor11showOutputsEv(i32 %arg) local_unnamed_addr addrspace(200) {
+; CHECK-LABEL: define {{[^@]+}}@_ZNK7KScreen6Doctor11showOutputsEv
+; CHECK-SAME: (i32 [[ARG:%.*]]) local_unnamed_addr addrspace(200) {
+; CHECK-NEXT:  entry:
+; CHECK-NEXT:    [[REF_TMP107:%.*]] = alloca [[CLASS_QSTRING:%.*]], align 16, addrspace(200)
+; CHECK-NEXT:    br label [[COND_TRUE:%.*]]
+; CHECK:       cond.true:
+; CHECK-NEXT:    [[TMP0:%.*]] = bitcast [[CLASS_QSTRING]] addrspace(200)* [[REF_TMP107]] to i8 addrspace(200)*
+; CHECK-NEXT:    [[TMP1:%.*]] = call i8 addrspace(200)* @llvm.cheri.bounded.stack.cap.i64(i8 addrspace(200)* [[TMP0]], i64 16)
+; CHECK-NEXT:    [[TMP2:%.*]] = bitcast i8 addrspace(200)* [[TMP1]] to [[CLASS_QSTRING]] addrspace(200)*
+; CHECK-NEXT:    [[TMP3:%.*]] = bitcast [[CLASS_QSTRING]] addrspace(200)* [[REF_TMP107]] to i8 addrspace(200)*
+; CHECK-NEXT:    [[TMP4:%.*]] = call i8 addrspace(200)* @llvm.cheri.bounded.stack.cap.i64(i8 addrspace(200)* [[TMP3]], i64 16)
+; CHECK-NEXT:    [[TMP5:%.*]] = bitcast i8 addrspace(200)* [[TMP4]] to [[CLASS_QSTRING]] addrspace(200)*
+; CHECK-NEXT:    switch i32 [[ARG]], label [[COND_END_SINK_SPLIT:%.*]] [
+; CHECK-NEXT:    i32 -1, label [[COND_END:%.*]]
+; CHECK-NEXT:    i32 0, label [[COND_END]]
+; CHECK-NEXT:    ]
+; CHECK:       cond.end.sink.split:
+; CHECK-NEXT:    br label [[COND_END]]
+; CHECK:       cond.end:
+; CHECK-NEXT:    [[REF_TMP109_SINK:%.*]] = phi [[CLASS_QSTRING]] addrspace(200)* [ [[TMP5]], [[COND_TRUE]] ], [ [[TMP2]], [[COND_TRUE]] ], [ null, [[COND_END_SINK_SPLIT]] ]
+; CHECK-NEXT:    [[CALL_I_I546:%.*]] = call [[CLASS_QSTRING]] addrspace(200)* @use([[CLASS_QSTRING]] addrspace(200)* [[REF_TMP109_SINK]])
+; CHECK-NEXT:    ret void
+;
+entry:
+  %ref.tmp107 = alloca %class.QString, align 16, addrspace(200)
+  br label %cond.true
+
+cond.true:
+  switch i32 %arg, label %cond.end.sink.split [
+  i32 -1, label %cond.end
+  i32 0, label %cond.end
+  ]
+
+cond.end.sink.split:
+  br label %cond.end
+
+cond.end:
+  %ref.tmp109.sink = phi %class.QString addrspace(200)* [ %ref.tmp107, %cond.true ], [ %ref.tmp107, %cond.true ], [ null, %cond.end.sink.split ]
```

Comment:
I'm surprised this is legal IR to be honest, though maybe the lesser of two evils

Reply:
Yeah I'm not quite sure what the original was before llvm-reduce. Maybe it looked a bit more sane.